### PR TITLE
[Merged by Bors] - rust 1.53.0 updates

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

`make lint` failing on rust 1.53.0.

## Proposed Changes

1.53.0 updates

## Additional Info

I haven't figure out why yet, we were now hitting the recursion limit in a few crates. So I had to add `#![recursion_limit = "256"]` in a few places

### CODE REVIEW

File: beacon_node/eth2_libp2p/src/peer_manager/mod.rs [hunk 1]
```diff
@@ -583,11 +581,7 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
             // port is removed, which is assumed to be associated with the discv5 protocol (and
             // therefore irrelevant for other libp2p components).
             let mut out_list = enr.multiaddr();
-            out_list.retain(|addr| {
-                addr.iter()
-                    .find(|v| matches!(v, MProtocol::Udp(_)))
-                    .is_none()
-            });
+            out_list.retain(|addr| !addr.iter().any(|v| matches!(v, MProtocol::Udp(_))));
```
Comment (michaelsproul) [inline beacon_node/eth2_libp2p/src/peer_manager/mod.rs:584]: Clippy's getting pretty clever! This looks right to me but I wouldn't mind some eyes on it from a network dev

File: beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs [hunk 2]
```diff
@@ -64,7 +64,6 @@ impl PeerSyncStatus {
 
     pub fn update(&mut self, new_state: PeerSyncStatus) -> bool {
         if *self == new_state {
-            *self = new_state;
```
Comment (divagant-martian) [inline beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: the problem here is that `PeerSyncStatus` uses an == that resembles a `matches!(..)`. So removing this line does change the logic

**Review by michaelsproul — APPROVED** [review]
LGTM, thanks for slogging through Clippy's complaints!

I think the `recursion_limit` change could be due to any number of things, possibly `tokio::select`, changes to `rustc`'s accounting for the limit, all our custom macros or some combination thereof. For a project of our size I think doubling the limit in a few places is fine, even if we haven't dug into the nitty gritty of exactly _why_.

Happy to merge!

File: beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs (hunk 2 above)
Comment (michaelsproul) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: Good catch! I was wondering if Clippy suggested this, but it seems it's just the generic `branches_sharing_code` lint that fires, and it's quite subtle to refactor correctly in this case.

Comment (michaelsproul) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: I had a go at restoring the previous functionality while dodging Clippy in f19be0de21c1ca52f85fabede1b685c78f0e2138 (seeing as Sean is afk).

Comment (divagant-martian) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: me approves

File: beacon_node/eth2_libp2p/src/peer_manager/mod.rs (hunk 1 above)
Comment (divagant-martian) [reply beacon_node/eth2_libp2p/src/peer_manager/mod.rs:584]: lgtm

File: beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs (hunk 2 above)
Comment (divagant-martian) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: tho I wouldn't mind changing this to use something different to equality. Hinders readability and doesn't really feel quite right in hindsight. But this is out of scope for the current PR. I'm fine with @michaelsproul 's solution :)

**Review by divagant-martian — APPROVED** [review]

File: beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs (hunk 2 above)
Comment (michaelsproul) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: Yeah, I agree it should be a separate method like `same_status`, `same_kind`, or something

**Comment by michaelsproul** [comment]
Merging. The `cargo-udeps` failure is just network flakiness

bors r+

**Comment by michaelsproul** [comment]
bors retry

**Comment by michaelsproul** [comment]
something's screwed with the cargo nightly, I'll fix it after lunch

**Comment by michaelsproul** [comment]
can't reproduce locally, trying it again

bors retry

**Comment by michaelsproul** [comment]
bors retry

**Comment by michaelsproul** [comment]
bors try

**Comment by michaelsproul** [comment]
bors try-

**Comment by michaelsproul** [comment]
It seems that the issue was `cargo udeps` using HTTP/2 multiplexing from the Github Actions box. I don't know what changed that caused it to fail suddenly, but it seems to be fixed by the `http.multiplexing = false` config for Cargo. At some point in the future we should try removing this hack to see if it's still necessary. It's _possible_ that we got severely unlucky and everything was actually caused by an underlying issue with Github's infra that resolved as soon as I tried this hack...

Anyway

bors r+

File: beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs (hunk 2 above)
Comment (realbigsean) [reply beacon_node/eth2_libp2p/src/peer_manager/peer_sync_status.rs:67]: thanks guys 🙂

**Pull request closed without merging**